lib/json/json.go

```diff
@@ -233,6 +239,27 @@ func encode(thread *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, k
 	return starlark.String(buf.String()), nil
 }
 
+func encodeIndent(thread *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	prefix, indent := "", "\t" // keyword-only
+	if err := starlark.UnpackArgs(b.Name(), nil, kwargs,
+		"prefix?", &prefix,
+		"indent?", &indent,
+	); err != nil {
+		return nil, err
+	}
+	// Rely on encode() to parse the positional parameter (since the signature matches); use our b so
+	// error messages are attributed to json.encode_indent.
+	str, err := encode(thread, b, args, nil)
+	if err != nil {
+		return nil, err
+	}
+	buf := new(bytes.Buffer)
```

Comment:
strings.Builder

will save an allocation.

Reply:
Oh, good point, the standard encoding/json package is indeed a constraint. This is fine.